### PR TITLE
Feature: Implements MapFields()

### PR DESCRIPTION
**Pull request by Paulholio**
This is an attempt at eventually implementing the feature request from:

[Issue 285: Updating All Fields](https://github.com/python-bonobo/bonobo/issues/285)

There is more work that needs to be done, but I think if we can spec out what types of operations we would want future users to be able to do across all fields, this can become very useful, especially on the validation and possibly security front. Let me know your opinions on this, please.

### CODE REVIEW

**Comment by hartym** [comment]
Hello,

Thanks for your contribution. A few things:

- Let's call this `MapFields`, to be named like python's `map()` builtin (but refers to "fields", as we could have a `Map` that does this logic on a whole row)
- Let's use the python callable to apply instead of a string : pass `str.upper` as parameter to the transformation, so it allows to use any callable the same way.
- Please add an unit test (even if not very complete) to check that this continue to work over time, and to check that edge cases work as expected (what if we get an empty row? what if we pass mixed type rows?)

Thank you, this is a nice one !

**Comment by Paulholio** [comment]
I have this renamed to MapFields and will have users pass in a function. Still need to write a unit test.

File: bonobo/nodes/basics.py [hunk 1]
```diff
@@ -314,6 +316,32 @@ def _Format(context, bag):
...
 
+@transformation_factory
+def MapFields(callFunc):
+    # TODO: Figure out how to more intelligently handle correctable errors
+    # Use this wrapper to call the function that we pass in and handle the error
+    # For now, handle the error by returning the unmodified value
+    def _wrapfunc(func, val):
+        try:
+            retval = func(val)  # Attempt to use the function
+        except:
```
Comment (hartym) [inline bonobo/nodes/basics.py]: We should never use a "catchall" except, instead, catch `Exception`.

Not sure this behaviour is wanted though, as it may come to unpredictable behaviour.

File: bonobo/nodes/basics.py [hunk 2]
```diff
@@ -314,6 +316,32 @@ def _Format(context, bag):
...
+            retval = val  # If we get any error, return the value unmodified
+
+        return retval
+
+    @use_raw_input
+    def _MapFields(bag):
+        nonlocal callFunc
+
+        b = {key: _wrapfunc(callFunc, value) for key, value in bag._asdict().items()}  # iterate the wrapped callable
+        makenamedtuple = collections.namedtuple('Bag', ' '.join((b.keys())))  # constructor of the returned tuple
```
Comment (hartym) [inline bonobo/nodes/basics.py]: You can use @use_context and get/set the input/output type from there. I think it's better than guessing the user would want this namedtuple.
